src/lib/ares_private.h

```diff
@@ -181,6 +184,8 @@ struct ares_query {
   size_t        timeouts;   /* number of timeouts we saw for this request */
   ares_bool_t   no_retries; /* do not perform any additional retries, this is
                              * set when a query is to be canceled */
+  ares_array_t *failed_servers_attempted; /* Failed servers attempted to be used
```

Comment:
why do we need to track this?  the prior logic would choose it in highest priority order.  failed servers wouldn't be probed until the timer expired for retrying them.